<launch>
    <param name="robot_description"
           command="$(find xacro)/xacro $(find ur_description)/urdf/ur5e.xacro" />

    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />

    <node pkg="tf" type="static_transform_publisher" name="world_to_base_broadcaster"
          args="0 0 0 0 0 0 world base_link 100" />

    <param name="use_sim_time" value="true"/>
    <rosparam>
        cam_config:
            eye_in_hand_camera:
                stream_type: 1
                frequency: 30
    </rosparam>
    <rosparam file="$(find ibvs_ur5_sim)/config/ur5_controllers.yaml" command="load"/>
    <rosparam>
        MujocoPlugins:
          - {type: mujoco_ros_control/MujocoRosControlPlugin, hardware: {type: mujoco_ros_control/DefaultRobotHWSim, control_period: 0.002}}
          - {type: mujoco_ros_mocap/MocapPlugin, body_name: "aruco_target", topic_name: "/mocap/pose", publish_tf: true}
          - {type: ibvs_ur5_sim/PseudoDepthPlugin}
    </rosparam>
    <param name="camera_name" type="str" value="eye_in_hand_camera"/>
    <param name="aruco_marker_name" type="str" value="aruco_target"/>
    <param name="aruco_marker_size" type="double" value="0.10"/>
    <node name="mujoco_server" pkg="mujoco_ros" type="mujoco_node" output="screen">
        <param name="modelfile" value="$(find ibvs_ur5_sim)/worlds/ur5e_ibvs_scene.xml" />
        <param name="headless" value="false"/>
    </node>
    <node name="controller_spawner" pkg="controller_manager" type="spawner" output="screen"
          args="joint_state_controller pos_joint_traj_controller"/>
    <node name="aruco_detector" pkg="ibvs_ur5_sim" type="aruco_detector_node.py" output="screen"/>
    <node name="initial_pose_setter" pkg="ibvs_ur5_sim" type="set_initial_pose.py" output="screen"/>
    <!-- <node name="ibvs_controller" pkg="ibvs_ur5_sim" type="ibvs_control_node.py" output="screen"/> -->
</launch>